model/riscv_insts_zce_ext.sail

```diff
@@ -0,0 +1,1021 @@
+/*  2. Redistributions in binary form must reproduce the above copyright                 */
+/*     notice, this list of conditions and the following disclaimer in                   */
+/*     the documentation and/or other materials provided with the                        */
+/*     distribution.                                                                     */
+/*                                                                                       */
+/*  THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS''                   */
+/*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED                    */
+/*  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A                      */
+/*  PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR                  */
+/*  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,                         */
+/*  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT                     */
+/*  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF                     */
+/*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND                  */
+/*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,                   */
+/*  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT                   */
+/*  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF                   */
+/*  SUCH DAMAGE.                                                                         */
+/*=======================================================================================*/
+
+/* ****************************************************************** */
+/* This file specifies the instructions in the 'ZCEE' extension.      */
+/* ****************************************************************** */
+
+/* ------------------------------------------------------------------ */
+/* Sail Model for the c.l and c.s encoding group                */
+/* ------------------------------------------------------------------ */
+
+/* ****************************************************************** */
+union clause ast = C_LBU : (bits(2), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ C0]
+mapping clause encdec_compressed = C_LBU(ui65, rs1, rd)
+  <-> 0b100 @ 0b000 @ rs1 : cregidx @ ui65 : bits(2) @ rd : cregidx @ 0b00
+
+function clause execute (C_LBU(uimm, rsc, rdc)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+
+  let rd = creg2reg_idx(rdc);
+  let rs = creg2reg_idx(rsc);
+
+  let rsext = X(rs) + immext;
+  X(rs) = rsext;
+
+   match execute(LOAD(imm, rs, rd, false, WORD, false, false)) {
+     RETIRE_SUCCESS => {
+                         X(rd) = EXTZ(X(rd)[7..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
```

Comment:
The existing C instructions (e.g. C_LW) do this in a simpler way - could just reuse that kind of logic?
```suggestion
mapping clause encdec_compressed = C_LBU(ui65, rs1, rd)
  <-> 0b100 @ 0b000 @ rs1 : cregidx @ ui65 : bits(2) @ rd : cregidx @ 0b00

function clause execute (C_LBU(uimm, rsc, rdc)) = {
  let imm : bits(12) = EXTZ(uimm);
  let rd = creg2reg_idx(rdc);
  let rs = creg2reg_idx(rsc);
  execute(LOAD(imm, rs, rd, false, BYTE, false, false))
```

Reply:
I did look at the implementation, but, according to my understanding of the specifications, this required the additional steps that i have implemented, could you please clarify my understanding.

Reply:
What additional/different steps do you think you need, and can you point at where in the specification it says you need them? I agree with Alex.

Reply:
I need to add the zero extended value of immediate to the contents of the source register and then put it into the destination, is my understanding from the spec.